authors/apps/profiles/tests/test_profiles.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import status
+
+from authors.apps.profiles.tests.basetests import BaseTest
+
+
+class TestViewProfile(BaseTest):
+    """
+    Tests for viewing profile
+    """
+
+    def test_successfully_getting_profile(self):
+        """
+        Test view profile successfully
+        """
+        self.is_authenticated("adam@gmail.com", "@Us3r.com")
+        response = self.get_profile_with_valid_username()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_getting_profile_with_invalid_username(self):
+        """
+        Test view profile for invalid username
+        """
+        self.is_authenticated("adam@gmail.com", "@Us3r.com")
+        response = self.get_profile_with_invalid_username()
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_profile_successfuly(self):
+        """
+        Test method for successfully updtaing a user
+        """
+        self.is_authenticated("adam@gmail.com", "@Us3r.com")
+        response = self.update_user_profile(
+            "Abraham", "Kamau", "I love history")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_profile_while_loggedout(self):
+        """
+        Test method for updating while user is not authenticated
+        """
+        response = self.update_user_profile(
```

Comment:
You could consider testing whether a user can update username. Also, you are only testing the status codes returned. Consider testing the messages returned too.

Reply:
@IssaIan There is no username field in the database, the username is gotten from the User table which, to Profiles is a ForeignKey hence what is being saved in the Profile is just but the User Id